src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BlockListener.java

```diff
@@ -58,8 +59,21 @@ public void onBlockPlaceExisting(BlockPlaceEvent e) {
         // This prevents Players from placing a block where another block already exists
         // While this can cause ghost blocks it also prevents them from replacing grass
         // or saplings etc...
-        if (BlockStorage.hasBlockInfo(e.getBlock())) {
-            e.setCancelled(true);
+        Block block = e.getBlock();
+
+        if (e.getBlockReplacedState().getType().isAir()) {
+            SlimefunItem sfItem = BlockStorage.check(block);
+
+            if (sfItem != null) {
+                for (ItemStack item : sfItem.getDrops()) {
+                    if (!item.getType().isAir()) {
+                        block.getWorld().dropItemNaturally(block.getLocation(), item);
+                    }
+                }
+                BlockStorage.clearBlockInfo(block);
+            } else {
+                e.setCancelled(true);
+            }
         }
```

Comment:
I would suggest to add a fallback here, because the `BlockPlaceEvent` will also be called when a Player tries to place the block in the same space as tall grass or other passable materials.
For these, we want to cancel it nonetheless as they would need to destroy the block first.

```suggestion
        } else if (BlockStorage.hasBlockInfo(e.getBlock())) {
            e.setCancelled(true);
        }
```